infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java

```diff
@@ -154,7 +152,7 @@ public void createOrReplace(PersonalAccessToken personalAccessToken)
                               "%s://%s:%s@%s%s",
                               scmUrl.getProtocol(),
                               getUsernameSegment(personalAccessToken),
-                              URLEncoder.encode(personalAccessToken.getToken(), UTF_8),
+                              personalAccessToken.getToken(),
```

Comment:
We have to know why previously it was done in that way.

Reply:
This was added a long time ago by @mshaposhnik, I don't think we can find the reason.

Reply:
Thanks @mshaposhnik for the explanation, looks like we would need to find a better solution